eox_core/edxapp_wrapper/backends/enrollment_h_v1.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Backend for the create_edxapp_user that works under the open-release/hawthorn.beta1 tag
+"""
+
+from __future__ import absolute_import, unicode_literals
+from enrollment import api
+from enrollment.errors import CourseModeNotFoundError
+from student.models import (
+    AlreadyEnrolledError,
+    CourseFullError,
+    EnrollmentClosedError,
+    NonExistentCourseError
+)
+from course_modes.models import CourseMode
+
+
+def create_enrollment(*args, **kwargs):
+
+    errors = []
+    user_id = kwargs.get('username')
+    course_id = kwargs.get('course_id')
+    mode = kwargs.get('mode')
+    is_active = kwargs.get('is_active', True)
+    enrollment_attributes = kwargs.get('enrollment_attributes', None)
+    validation_errors = check_edxapp_enrollment_is_valid(*args, **kwargs)
```

Comment:
What do you think of moving this into the serializer validation method? This way if the `check_edxapp_enrollment_is_valid` fails we inmediately return from the serializer in the class EdxappEnrollment#L60

Reply:
Serializer validation method already calls this method too. This way of doing in both part is what /v1/users/ does.